### ISSUE TITLE
Cannot Load Saved Shakespeare Model

### ISSUE THREAD

**Issue by adrianlyjak** [tflearn/tflearn#18]
Running the `lstm_generator_shakespeare.py` example from a saved model is giving me the following stacktrace:

```
  File "lstm_generator_shakespeare.py", line 37, in <module>
    m.load('model_shakespeare-5')
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tflearn/models/generator.py", line 226, in load
    self.predictor = Evaluator(self.net, model=model_file)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tflearn/helpers/evaluator.py", line 26, in __init__
    self.graph = tensors[0].graph
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/ops/array_ops.py", line 154, in _SliceHelper
    sliced = slice(tensor, indices, sizes)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/ops/array_ops.py", line 204, in slice
    return gen_array_ops._slice(input_, begin, size, name=name)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/ops/gen_array_ops.py", line 908, in _slice
    name=name)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/ops/op_def_library.py", line 655, in apply_op
    op_def=op_def)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2042, in create_op
    set_shapes_for_outputs(ret)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1528, in set_shapes_for_outputs
    shapes = shape_func(op)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/ops/array_ops.py", line 778, in _SliceShape
    input_shape.assert_has_rank(ndims)
  File "/Users/adrianlyjak/dev/futz/tflearn/examples/nlp/lib/python2.7/site-packages/tensorflow/python/framework/tensor_shape.py", line 596, in assert_has_rank
    raise ValueError("Shape %s must have rank %d" % (self, rank))
ValueError: Shape (?, 67) must have rank 1
```

It seems to be coming from line 26 in `helpers/evaluator.py`  `Evaluator.__init__`
```python
 self.graph = tensors[0].graph
```

Here, `tensors` is already a a `Tensor`, so changing the line to `self.graph = tensors.graph` fixes the slice error. Is `tensors` ever another type?

**Comment by aymericdamien** [comment]
Hi, thanks for reporting the issue!

There is a strange behavior here, 
```
self.tensors = to_list(tensors)
self.graph = tensors[0].graph
```
to_list should ensure that tensors is a list... thus tensors[0] should be defined (or throw an index exception if None). 'tensors' is a list of tensors (because Evaluator can evaluate multiple tensors at the same time). In case the user send a single Tensor, then tensors is converted to an array using 'to_list' (and that array then only containing a single tensor). But here, it seem it didn't work.

As a fix, changing load function of generator class (https://github.com/tflearn/tflearn/blob/master/tflearn/models/generator.py#L215) to will fix that issue:
```
def load(self, model_file):
        """ Load.

        Restore model weights.

        Arguments:
            model_file: `str`. Model path.

        """
        self.trainer.restore(model_file)
        self.session = self.trainer.session
        self.predictor = Evaluator([self.net],
                                   session=self.session)
```

But furthermore investigation are needed to find out why to_list haven't converted the Tensor to a `list` of Tensor.

**Comment by adrianlyjak** [comment]
Aha! No, I don't think that fix will work. The problem is that your still referencing the possible tensor (`tensor`), not the list (`self.tensor`). New fix in #23 works for me
